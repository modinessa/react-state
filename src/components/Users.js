<<<<<<< HEAD
import React from "react";
import * as constants from "../constants/constants.js";

export function Users({users}) {
	return (
		<div className="app-section__users ">
			{users.map((user) => (
				<div key={user.id} className = {constants.USER_CARD_CLASS}>
					<img src={user.avatar} className="app-section__member-img"
							alt="User"/>
					<p className="app-section__member-review">
						{constants.USER_REVIEW}
					</p>
					<div className="app-section__member-name">
						{user.firstName} {user.lastName}
					</div>
					<div className="app-section__member-position">
						{user.position}
					</div>
			</div>
			))}
		</div>
	)
}
=======
		import React from "react";
		import * as constants from "../constants/constants.js";
		
		export function Users({users}) {
			return (
				<div className="app-section__users ">
					{users.map((user) => (
					<div key={user.id} className = {constants.USER_CARD_CLASS}>
						<img src={user.avatar} className="app-section__member-img"
								alt="User"/>
						<p className="app-section__member-review">
							{constants.USER_REVIEW}
						</p>
						<div className="app-section__member-name">
							{user.firstName} {user.lastName}
						</div>
						<div className="app-section__member-position">
							{user.position}
						</div>
					</div>
					))}
				</div>
			)
		}
>>>>>>> 745986d467c6bcca5bf7155bb0a99a7a77a5e364
