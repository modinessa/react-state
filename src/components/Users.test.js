import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom'; 

import {Users} from './Users';

const TEST_USER = {
	id: 'fake_id',
	firstName: 'first',
	lastName: 'last',
	position: 'test',
	avatar: 'fake_url',
};

describe('Users', () => {
	it('renders component without users', async () => {
		expect(renderer.create(
			<BrowserRouter>
				<Users users={[]}/>
			</BrowserRouter>)).toMatchSnapshot();
		});

	it('renders component with users', () => {
		expect(renderer.create(
			<BrowserRouter>
				<Users users={[TEST_USER]}/>
			</BrowserRouter>)).toMatchSnapshot();
	});
});
