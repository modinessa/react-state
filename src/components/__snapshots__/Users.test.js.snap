// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users renders component with users 1`] = `
<div
  className="app-section__users "
>
  <div
    className="app-section__member-card app-section__member-card-select"
  >
    <img
      alt="User"
      className="app-section__member-img"
      src="fake_url"
    />
    <p
      className="app-section__member-review"
    >
      PLACEHOLDER: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do incididunt ut labore et dolor.
    </p>
    <a
      className="user-link"
      href="/community/fake_id"
      onClick={[Function]}
    >
      <div
        className="app-section__member-name"
      >
        first
         
        last
      </div>
    </a>
    <div
      className="app-section__member-position"
    >
      test
    </div>
  </div>
</div>
`;

exports[`Users renders component without users 1`] = `
<div
  className="app-section__users "
/>
`;
